src/Poseidon/Package.hs

```diff
@@ -516,6 +517,15 @@ getJointGenotypeData showAllWarnings intersect pacs maybeSnpFile = do
         (Just snp, Nothing) -> unless intersect $ yield (snp, V.replicate n Missing)
         _ ->  return ()
 
+joinEntryPipe :: (MonadIO m) => Bool -> [Int] -> [String] -> Pipe [Maybe (EigenstratSnpEntry, GenoLine)] (EigenstratSnpEntry, GenoLine) m r
+joinEntryPipe showAllWarnings nrInds pacNames = for cat $ \maybeEntries -> do
+    eitherJE <- liftIO . try $ joinEntries showAllWarnings nrInds pacNames maybeEntries
+    case eitherJE of
+        Left (PoseidonGenotypeException err) ->
+            when showAllWarnings . liftIO . hPutStrLn stderr $ "Skipping SNP due to " ++ err
```

Comment:
I'm glad you ask. So, the issue here is that incongruent SNPs (other than incongruent SNP IDs or genetic positions) lead to a failure in the join operation, so I can't just put a warning there, as I do with incongruent SNP IDs or so, but have to throw an exception and catch it here during the stream-processing. But of course, a stream-map operation that can skip stream-items is not a simple `P.mapM` anymore, as we had it before, but an actual pipe that can also decide which entries to forward and which to skip. That's done using the `for cat`, which is a pipe and I can manually say in which cases I `yield` and in which cases not.

So the general behaviour is: incongruent SNPs will always be skipped. With the `-w` flag, you also get a warning, but the skipping cannot be turned off. 

I'm happy to add another flag to control this behaviour, but I think the always-skipping policy should work for 99% of cases.

Reply:
Brilliant! Thank you so much for the explanation! 
Maybe this could be added as a comment to the function for future reference? Maybe something like:
```
-- removes incongruent SNPs from the stream
-- if showAllWarnings == True, then a warning is printed,
-- otherwise they are dropped silently
```

Reply:
Ah yes. Certainly should do a better job with code comments. Just added.